### ISSUE TITLE
Overzealous IDE0032 "Use auto property" when using the same backing field for two properties

### ISSUE THREAD

**Issue by Tragetaschen** [dotnet/roslyn#45384]
**Version Used**: 

16.6.2

**Steps to Reproduce**:
```C#
public class Test
{
    private readonly string serialNumber; // IDE0032
    public Test(string serialNumber) => this.serialNumber = serialNumber;
    public string DeviceSerialNumber => serialNumber;
    public string SensorSerialNumber => serialNumber;
}
```

**Expected Behavior**:
The same backing field is used for two properties. It's not a clear which property should be an auto property, so IDE0032 should not be triggered.

**Actual Behavior**:
The code fix chooses the first property in source code order and converts the second property to reference the first. While functionally unchanged, in my opinion, this makes the code less readable as it's not clear anymore that both properties are sourced by the same value.

![image](https://user-images.githubusercontent.com/1947968/85391211-2b7c5f80-b54a-11ea-87d5-7b19aef4646c.png)

**Comment by CyrusNajmabadi** [comment]
I think the current behavior is fine.  The IDE is just telling the user this is possible.  The user is free to decide if they want to do this or not.

**Comment by mavasani** [comment]
I agree with @CyrusNajmabadi. This is by design, unless we get lot of feedback that this behavior is undesirable or needs some option.

**Comment by Tragetaschen** [comment]
Here are my thoughts:

I usually like the Use auto property analyzer and have configured it to show as error in the IDE.
As outlined in the OP, the fixer chooses one of the two properties to be an auto property and references it from the other, which breaks source code symmetry. "Source code order" is used as a tie breaker which seems a bit arbitrary.

That being said, my use case yesterday was the first time I encountered it for a long time using the analyzer. I could live with disabling it just this once in the source code.

**Comment by CyrusNajmabadi** [comment]
hi @Tragetaschen.  Thanks for the info.  Our recommendation in cases like this (where 99.9% of the time you like it, and 0.1% of the time you don't), is to just add a suppression in that single case.  It's just too much of a corner case to add special support/options for.
